payments/stripe/__init__.py

```diff
@@ -76,3 +94,141 @@ class StripeCardProvider(StripeProvider):
+    tax_code: Optional[str] = field(init=False, repr=False, default=None)
+
+
+@dataclass
+class StripePriceData:
+    currency: str
+    product_data: StripeProductData
+    unit_amount: int
+    recurring: Optional[dict] = field(init=False, repr=False, default=None)
+    tax_behavior: Optional[str] = field(init=False, repr=False, default=None)
+
+
+@dataclass
+class StripeLineItem:
+    price_data: StripePriceData
+    quantity: int
+    adjustable_quantity: Optional[dict] = field(init=False, repr=False, default=None)
+    dynamic_tax_rates: Optional[dict] = field(init=False, repr=False, default=None)
+    tax_rates: Optional[str] = field(init=False, repr=False, default=None)
+
+
+class StripeProviderV3(BasicProvider):
+    """Provider backend using `Stripe <https://stripe.com/>` api version 3_.
+
+    This backend does not support fraud detection.
+
+    :param api_key: Secret key assigned by Stripe.
+    :param payment_method_types: From Stripe API, comma separated
+    :param use_token: Use instance.token instead of instance.pk in client_reference_id
+    """
+
+    form_class = PaymentFormV3
+
+    def __init__(
+        self,
+        api_key,
+        use_token=True,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.api_key = api_key
+        self.use_token = use_token
+
+    def get_form(self, payment, data=None):
+        if not payment.transaction_id:
+            try:
+                session = self.create_session(payment)
+            except PaymentError as pe:
+                payment.change_status(PaymentStatus.ERROR, str(pe))
+                raise PaymentError(pe)
+            else:
+                payment.attrs.session = session
+                payment.transaction_id = session.get("id", None)
+                payment.save()
+
+        if "url" not in payment.attrs.session:
+            raise PaymentError(_("No `url` in payment.attrs.session"))
+
+        raise RedirectNeeded(payment.attrs.session.get("url"))
+
+    def create_session(self, payment):
+        """Makes the call to Stripe to create the Checkout Session"""
+        if not payment.transaction_id:
+            stripe.api_key = self.api_key
+            session_data = {
+                "line_items": self.get_line_items(payment),
+                "mode": "payment",
+                "success_url": payment.get_success_url(),
+                "cancel_url": payment.get_failure_url(),
+                "client_reference_id": payment.token if self.use_token else payment.pk,
+            }
+            # Patch session with billing email if exists
+            if payment.billing_email:
+                session_data.update({"customer_email": payment.billing_email})
+            try:
+                session = stripe.checkout.Session.create(**session_data)
+            except stripe.error.StripeError as e:
+                # Payment has been declined
+                raise PaymentError(e)
+            else:
+                return session
```

Comment:
Unnecessary else:

```suggestion
            try:
                return stripe.checkout.Session.create(**session_data)
            except stripe.error.StripeError as e:
                # Payment has been declined
                raise PaymentError(e)
```

Reply:
OK, i'll make the change, but it's a pattern I don't use it (a return inside a try, it messes with my mind)

Reply:
Since you're the one working on the Stripe provider, if you find the shorter version confusing you can leave it as it is.

FWIW, you can think of 

```
return somefunc(y)
```

As equivalent to:

```
x = somefunc(y)
return x
```

The return is infallible, but the function call _can_ fail.